File: app/src/main/java/ru/divizdev/coinrate/presentation/detail/presenter/CoinRateDetailView.java
package ru.divizdev.coinrate.presentation.detail.presenter;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

import ru.divizdev.coinrate.entities.CoinRateUI;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface CoinRateDetailView extends MvpView {

    void setCurrencyFrom(String currencyFrom);

    void setCurrencyTo(String currencyTo);

    void setValueTo(String value);

    void clearAll();

    void showError(String error);

    void setData(CoinRateUI coinRateUI);

}

File: app/src/main/java/ru/divizdev/coinrate/presentation/detail/presenter/CoinRateDetailView.kt
package ru.divizdev.coinrate.presentation.detail.presenter

import moxy.viewstate.strategy.StateStrategyType
import moxy.viewstate.strategy.AddToEndSingleStrategy
import moxy.MvpView
import ru.divizdev.coinrate.presentation.entities.CoinRateUI

@StateStrategyType(AddToEndSingleStrategy::class)
interface CoinRateDetailView : MvpView {
    fun setCurrencyFrom(currencyFrom: String?)
    fun setCurrencyTo(currencyTo: String?)
    fun setValueTo(value: String?)
    fun clearAll()
    fun showError(error: String?)
    fun setData(coinRateUI: CoinRateUI?)
}